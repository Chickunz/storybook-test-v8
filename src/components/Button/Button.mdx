import { Meta, Canvas, ArgTypes } from "@storybook/blocks";
import * as ButtonStories from "./Button.stories";

<Meta of={ButtonStories} />

<style>
  {`
    .subheading {
      --mediumdark: '#999999';
      font-weight: 700;
      font-size: 13px;
      color: #999;
      letter-spacing: 6px;
      line-height: 24px;
      text-transform: uppercase;
      margin-bottom: 12px;
      margin-top: 40px;
    }
  `}
</style>

# Button Component

Making My Components Here

<div className="subheading">Button</div>

<div>This is using the primary button!</div>

<Canvas of={ButtonStories.Primary} />

<ArgTypes of={ButtonStories} />
