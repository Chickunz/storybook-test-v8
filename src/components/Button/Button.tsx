import React from "react";
import { Button as MUIButton, useForkRef } from "@mui/material";

export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  /**
   * Is this the principal call to action on the page?
   */
  primary?: boolean;
  /**
   * How large should the button be?
   */
  size?: "small" | "medium" | "large";
  /**
   * Button contents
   */
  label: string;
  /**
   * Optional click handler
   */
  onClick(): void;
}

export const Button = React.forwardRef(
  ({ primary, size, label, onClick }: ButtonProps, ref) => {
    const buttonRef = React.useRef<HTMLButtonElement>(null);
    const forkedRef = useForkRef(buttonRef, ref);

    return (
      <MUIButton onClick={onClick} ref={forkedRef}>
        {label}
      </MUIButton>
    );
  }
);

Button.displayName = "Button";

export default Button;
