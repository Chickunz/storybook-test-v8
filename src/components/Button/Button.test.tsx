import React from "react";
import Button from "./Button";
import { describe, it, expect, beforeEach } from "vitest";
import { render } from "@testing-library/react";

describe("Button Test Cases", () => {
  let getByRole: (arg0: string) => HTMLElement,
    asFragment: () => DocumentFragment;

  beforeEach(() => {
    ({ getByRole, asFragment } = render(
      <Button label={"Testing"} onClick={() => console.log("Hey")} />
    ));
  });

  it("Should Match Button Snapshot", () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
