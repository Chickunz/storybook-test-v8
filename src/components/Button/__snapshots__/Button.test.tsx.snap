// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Test Cases > Should Match Button Snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Testing
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
